cell/errors.go

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cell
+
+type DuplicateFlagError struct {
```

Comment:
This is fine. IMO, the most idiomatic way would be to have 

```
var ErrDuplicateFlag = errors.New("duplicate flag")
```

and then have the specific errors emitted wrap this sentinel error.

```
err = fmt.Errorf("'%s' declared twice: %w", flag.Name, ErrDuplicateFlag)
```

As a consequence, clients/tests can then match on this sentinel error.
```
if errors.Is(err, ErrDuplicateFlag) {}
```

Reply:
Aha, I knew I was doing something wrong. This resolves the testing package issue too.